plugins/imjournal/imjournal.c

```diff
@@ -413,6 +415,13 @@ readjournal(void)
 		tv.tv_usec = timestamp % 1000000;
 	}
 
+	/* save journal cursor (at this point we can be sure it is valid) */
+	sd_journal_get_cursor(j, &c);
```

Comment:
This is a potential performance issue.  I have not had a chance to verify the overhead of this.  It is on my plate to do.

Reply:
Obtaining cursor involves memory allocation, but otherwise should be cheap. All metadata that cursor is composed of are very likely in memory already. 

IOW, pointer dereferences inside ```sd_journal_get_cursor()``` shouldn't cause major page fault because ```EntryObject```  was loaded into memory previously since the plugin calls ```sd_journal_get_realtime_usec()``` anyway.

There is one more dereference outside of ```EntryObject```, i.e. fetching journal file ```seqnum_id```, that could potentially cause disk round-trip. However, ```seqnum_id``` will be accessed for every message in current file so it is unlikely that kernel would decide to swap out the page back to disk.

Reply:
All great in theory, best to test and measure overhead to be sure.

Reply:
@portante what do you require in order to be satisfied that this proposed PR does not cause a noticeable performance regression?  Do you have a test rig ready to go, and just need a rhel7.4 build with this patch in it?  Do you need other resources (manpower, computing)?